src/components/Courses/components/CourseCard/tests/courseCard.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { TAuthor, TCourse, TUser } from 'src/shared.types';
+import { CourseCard } from '../CourseCard';
+import { toHoursAndMinutes } from 'src/helper';
+import dayjs from 'dayjs';
+import { useCourseById } from 'src/hooks';
+
+const mockedState = {
+	user: { value: { token: 'token' } } as { value: TUser; isLoaded: boolean },
+	courses: {
+		value: [
+			{
+				id: '1',
+				title: 'title',
+				description: 'description',
+				duration: 120,
+				authors: ['1', '3'],
+				creationDate: '10/09/2023',
+			},
+		],
+		isLoaded: true,
+	} as { value: TCourse[]; isLoaded: boolean },
+	authors: {
+		value: [
+			{ id: '1', name: 'a1' },
+			{ id: '2', name: 'a2' },
+			{ id: '3', name: 'a3' },
+		],
+		isLoaded: true,
+	} as { value: TAuthor[]; isLoaded: boolean },
+};
+
+const mockedStore = {
+	getState: () => mockedState,
+	subscribe: jest.fn(),
+	dispatch: jest.fn(),
+};
+
+const mockedUsedNavigate = jest.fn();
+
+jest.mock('react-router-dom', () => ({
```

Comment:
Usually, we never mock such kind of libraries in real tests. Please use next approach instead:
```jsx
// 1. Create a reusable function which will render your component with any necessary wrappers, e.g.:
import { MemoryRouter, Route, Routes } from 'react-router-dom';
function renderComponent(element, initialState) {
	const elementToRender = (
		<Provider store={initialState}>
			<MemoryRouter>
				<Routes>
					<Route
						path='/'   <---- any path you need 
						element={
							<ComponentWithoutRouter>{children}</ComponentWithoutRouter>
						}
					></Route>
				</Routes>
			</MemoryRouter>
		</Provider>
	)
	return render(elementToRender)
}

// 2. use this function in your tests, e.g.:
renderComponent(<CourseCard courseId={'1'} />, mockedState);
```


This reusable utility function can be moved to a separate file (e.g.: src/test-utils.tsx) and imported by all your tests.